core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -127,33 +128,79 @@ object Battle {
+        
+        var captureChance = 10 + attacker.getAttackingStrength().toFloat() / defender.getDefendingStrength().toFloat() * 40
+        if (captureChance > 80) captureChance = 80f
+        if (100 * Random().nextFloat() > captureChance) return
+        
+        val newUnit = attacker.getCivInfo().placeUnitNearTile(defender.getTile().position, defender.getName())
+        if (newUnit == null) return // silently fail
+        attacker.getCivInfo().addNotification("Your [${attacker.getName()}] captured an enemy [${defender.getName()}]", newUnit.getTile().position, NotificationIcon.War)
+        
+        newUnit.currentMovement = 0f
+        newUnit.health = 50
     }
 
     private fun takeDamage(attacker: ICombatant, defender: ICombatant) {
-        var damageToDefender = BattleDamage.calculateDamageToDefender(attacker, attacker.getTile(), defender)
-        var damageToAttacker = BattleDamage.calculateDamageToAttacker(attacker, attacker.getTile(), defender)
+        var potentialDamageToDefender = BattleDamage.calculateDamageToDefender(attacker, attacker.getTile(), defender)
+        var potentialDamageToAttacker = BattleDamage.calculateDamageToAttacker(attacker, attacker.getTile(), defender)
+        
+        var damageToAttacker = potentialDamageToAttacker
+        var damageToDefender = potentialDamageToDefender
 
         if (defender.getUnitType().isCivilian() && attacker.isMelee()) {
             captureCivilianUnit(attacker, defender as MapUnitCombatant)
         } else if (attacker.isRanged()) {
-            defender.takeDamage(damageToDefender) // straight up
+            defender.takeDamage(potentialDamageToDefender) // straight up
         } else {
             //melee attack is complicated, because either side may defeat the other midway
             //so...for each round, we randomize who gets the attack in. Seems to be a good way to work for now.
 
-            while (damageToDefender + damageToAttacker > 0) {
-                if (Random().nextInt(damageToDefender + damageToAttacker) < damageToDefender) {
-                    damageToDefender--
+            while (potentialDamageToDefender + potentialDamageToAttacker > 0) {
+                if (Random().nextInt(potentialDamageToDefender + potentialDamageToAttacker) < potentialDamageToDefender) {
+                    potentialDamageToDefender--
                     defender.takeDamage(1)
                     if (defender.isDefeated()) break
                 } else {
-                    damageToAttacker--
+                    potentialDamageToAttacker--
                     attacker.takeDamage(1)
                     if (attacker.isDefeated()) break
                 }
             }
         }
+        
+        damageToAttacker -= potentialDamageToAttacker // We first set this to the maximal damage to be done, and now we
+        damageToDefender -= potentialDamageToDefender // subtract the amount of damage remaining. This yields the damage done
```

Comment:
U wot m8t
You realize this means that for ranged attacks, both values will be 0, yes?

```

        var damageToAttacker = potentialDamageToAttacker
        var damageToDefender = potentialDamageToDefender


            defender.takeDamage(potentialDamageToDefender) // straight up


        damageToAttacker -= potentialDamageToAttacker // We first set this to the maximal damage to be done, and now we
        damageToDefender -= potentialDamageToDefender // subtract the amount of damage remaining. This yields the damage done
```

I'm not sure what you tried to do here, but this is probably not the way.